source/common/upstream/maglev_lb.cc

```diff
@@ -46,8 +46,7 @@ MaglevTable::MaglevTable(const HostsPerLocality& hosts_per_locality,
       const std::string& address = host->address()->asString();
       table_build_entries.emplace_back(host, HashUtil::xxHash64(address) % table_size_,
                                        (HashUtil::xxHash64(address, 1) % (table_size_ - 1)) + 1,
-                                       max_host_weight > 0 ? effective_weight(host->weight(), i)
-                                                           : 0);
+                                       effective_weight(host->weight(), i));
```

Comment:
For my sanity, can you add an `ASSERT(max_host_weight > 0)` above this loop?

Reply:
Slightly more expensive but more direct alternative: just `ASSERT(host_weight != 0)` in the `effective_weight` lambda definition. This would catch the case where a single host has a surprising zero weight. Which do you prefer?

Reply:
... While we're at it, I could be extra defensive and assert that the locality weight is non-zero as well.

Reply:
`ASSERT` doesn't count performance-wise as it's debug-only, so whatever you think documents the property in the most direct way.